torchserve/datadog_checks/torchserve/inference_api_check.py

```diff
@@ -0,0 +1,34 @@
+# (C) Datadog, Inc. 2023-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from urllib.parse import urljoin
+
+from datadog_checks.base import AgentCheck
+
+
+class TorchserveInferenceAPICheck(AgentCheck):
+    __NAMESPACE__ = 'torchserve.inference_api'
+    SERVICE_CHECK_NAME = 'health'
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.tags = []
+        self.check_initializations.append(self.parse_config)
+
+    def parse_config(self):
+        self.tags = self.instance.get("tags", [])
+        self.tags.append(f"inference_api_url:{self.instance['inference_api_url']}")
+
+    def check(self, _):
+        ping_url = urljoin(self.instance['inference_api_url'], 'ping')
+
+        try:
+            self.log.debug("Querying URL: [%s]", ping_url)
+            response = self.http.get(ping_url)
+            self.log.debug("Inference API `response`: [%s]", response)
+            response.raise_for_status()
+        except Exception:
```

Comment:
I'm not sure what would be the benefit to have two different `except` blocks here since in both cases we are going to log the error and send a service check to critical

Reply:
Catching specific exception first is a perceived best practice for exception handling. [This](https://www.learnpython.dev/03-intermediate-python/40-exceptions/70-best-practices/) for example

It is always good to handle an expected exception, meaning the error path is understood and mitigated. In this case it is clear that raise for status will throw http error, so specifying the expected failure seems relevant

Reply:
Do I understand correctly you say that we should not send a critical service check if this is unexpected?

Reply:
Something like this? 

```
        except HTTPError as e:
            self.log.debug("An HTTPError occurred. %s", e, exc_info=True)
            self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.CRITICAL, tags=self.tags)
            raise
        except Exception as e:
            self.log.debug("An unexpected error occurred. %s", e, exc_info=True)
            self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.CRITICAL, tags=self.tags)
            raise
```

Reply:
updated

Reply:
Tests are failing because under the hood our http connector can raise a variety of exceptions (example [here](https://github.com/DataDog/integrations-core/blob/7a0397917c7842caa0aa38f9240c4c0d2d31ebc8/datadog_checks_base/datadog_checks/base/utils/http.py#L423-L426), but also ConnectionError in case of timeouts) which are not HTTPError. Moving it back to Exception

Reply:
TBH, I prefer to stick with Exceptions even if it's a bit broad to stay on the safe side and make sure we won't have any other types falling through the cracks, which could lead us to have to keep updating the list of exceptions and re-release the integration.

Reply:
Please add a debug log in the exception handler so that we can be aware of the exception

Reply:
The exception is already logged. For example:

```
 Error: HTTPConnectionPool(host='localhost', port=8083): Max retries exceeded with url: /ping (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0xffff82bc2b80>: Failed to establish a new connection: [Errno 111] Connection refused'))
      Traceback (most recent call last):
        File "/opt/datadog-agent/embedded/lib/python3.9/site-packages/urllib3/connection.py", line 174, in _new_conn
          conn = connection.create_connection(
        File "/opt/datadog-agent/embedded/lib/python3.9/site-packages/urllib3/util/connection.py", line 95, in create_connection
          raise err
        File "/opt/datadog-agent/embedded/lib/python3.9/site-packages/urllib3/util/connection.py", line 85, in create_connection
          sock.connect(sa)
      ConnectionRefusedError: [Errno 111] Connection refused
```

I can add a debug log though